libraries/hwlib/library/targets/usart_queue.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * =====================================================================================
+ *
+ *       Filename:  usart_queue.hpp
+ *
+ *    Description:  
+ *
+ *        Version:  1.0
+ *        Created:  14-03-19 11:25:21
+ *       Revision:  none
+ *       Compiler:  gcc
+ *
+ *         Author:  YOUR NAME (), 
+ *   Organization:  
+ *
+ * =====================================================================================
+ */
+#ifndef QUEUE_HPP
```

Comment:
The name for this specific file is a bit off since the file tells us it is only used for the usart but it looks like a general queue class